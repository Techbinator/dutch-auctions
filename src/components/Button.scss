@import "../config/scss/variables.scss";

button {
  color: #ffffff;
  font-size: 14px;
  line-height: 1.75;
  font-weight: 500;
  background-color: $primary;
  cursor: pointer;
  padding: 10px 0;
  width: 100%;
  &.secondary {
    background-color: $secondary;
  }
  &:hover {
    opacity: 0.95;
  }
}
