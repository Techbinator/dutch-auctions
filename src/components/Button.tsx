import React from "react";
import "./Button.scss";
interface IButton {
  type: "primary" | "secondary";
  text: string;
}

export default ({ type, text }: IButton) => (
  <button className={type}>{text}</button>
);
