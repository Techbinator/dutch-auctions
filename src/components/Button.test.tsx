import React from "react";
import Button from "./Button";
import { shallow } from "enzyme";

describe("Button component", () => {
  it("should render snapshot with default values", () => {
    const wrapper = shallow(<Button text="test" type="primary" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render disabled button with default values", () => {
    const wrapper = shallow(
      <Button disabled text="secondary" type="secondary" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
