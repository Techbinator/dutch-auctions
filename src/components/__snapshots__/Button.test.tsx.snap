// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render disabled button with default values 1`] = `
<button
  className="secondary"
  disabled={true}
>
  secondary
</button>
`;

exports[`Button component should render snapshot with default values 1`] = `
<button
  className="primary"
  disabled={false}
>
  test
</button>
`;
